p2p/discover/v5_udp.go

```diff
@@ -504,6 +506,9 @@ func (t *UDPv5) verifyResponseNode(c *callV5, r *enr.Record, distances []uint, s
 		return nil, errors.New("duplicate record")
 	}
 	seen[node.ID()] = struct{}{}
+	if t.nodeFilter != nil && !t.nodeFilter(node) {
```

Comment:
Since we’ve added node filtering in handleAddNode, what additional cases does this change cover?

Reply:
`handleAddNode` is table-level validation, `verifyResponseNode` is UDP protocol-level
  validation, this ensures only matching peers can respond to`Findnode` request.